File: csuro_intro_cpp/src/nodo_pub.cpp
#include "ros/ros.h"
#include "std_msgs/Int64.h"

int main(int argc, char **argv)
{

   ros::init(argc, argv, "num_publisher");
   ros::NodeHandle n;

   ros::Publisher num_pub = n.advertise<std_msgs::Int64>("/message", 1);

   ros::Rate loop_rate(10);

   int count=0;

   while (ros::ok())
   {

     std_msgs::Int64 msg;
     msg.data = count++;

     num_pub.publish(msg);

     ros::spinOnce();
     loop_rate.sleep();
   }

   return 0;

 }

File: csuro_intro_py/scripts/nodo_pub.py
#!/usr/bin/env python

import rospy
from std_msgs.msg import Int64

if __name__ == '__main__':
    try:
        rospy.init_node('num_publisher', anonymous=False)
        
        pub = rospy.Publisher('/message', Int64, queue_size=1)
        
        rate = rospy.Rate(10)
        count = 0;
        
        while not rospy.is_shutdown():
        	
        	count = count + 1
        	pub.publish(count)
        
        	rate.sleep()
        	
        
    except rospy.ROSInterruptException:
        pass